iceoryx_utils/include/iceoryx_utils/internal/concurrent/periodic_task.hpp

```diff
@@ -33,13 +33,22 @@ template <typename T>
 class PeriodicTask
 {
   public:
+    /// @brief Creates a periodic task. The specified callable is stored but not executed.
```

Comment:
Please add some documentation regarding the behaviour of the `PeriodicTask`. This is a lessons learned from the `posix::timer`, back then we called it:
```
    enum class CatchUpPolicy
    {
        SKIP_TO_NEXT_BEAT,
        IMMEDIATE,
        TERMINATE
    };
```

Reply:
will add a note that it behaves like `IMMEDIATE`. There is a [proposal](https://github.com/eclipse-iceoryx/iceoryx/issues/337#issuecomment-760909671) to implement that feature for the `posix::Timer` refactoring which will then be used in the `PeriodicTask`

Reply:
@mossmaurice well, apparently it's currently not `IMMEDIATE` but more like `EXTEND`, which means the time the callback takes for execution is not taken into account for simplicity reasons. This will be added once the previous mentioned refactoring is done.